File: demo/src/main/java/com/line/demo/yu/GradientStrokeSpan.java
package com.line.demo.yu;

import android.graphics.LinearGradient;
import android.graphics.Shader;
import android.text.TextPaint;
import android.text.style.CharacterStyle;
import android.text.style.UpdateAppearance;

public class GradientStrokeSpan extends CharacterStyle implements UpdateAppearance {

    private float mStrokeWidth = 3f;
    private int[] mGradientColors;

    public GradientStrokeSpan(float strokeWidth, int[] gradientColors) {
        this.mStrokeWidth = strokeWidth;
        this.mGradientColors = gradientColors;
    }

    @Override
    public void updateDrawState(TextPaint paint) {
//        paint.setStyle(Paint.Style.STROKE);
        paint.setStrokeWidth(mStrokeWidth);
        LinearGradient gradient = new LinearGradient(0, 0, paint.measureText("Sample Text"), 0,
                mGradientColors, null, Shader.TileMode.CLAMP);
        paint.setShader(gradient);
    }



}

File: demo/src/main/java/com/line/demo/yu/round/GradientStrokeSpan.kt
package com.line.demo.yu.round

import android.graphics.Color
import android.graphics.LinearGradient
import android.graphics.Paint
import android.text.style.CharacterStyle
import android.text.style.UpdateAppearance
import android.text.TextPaint
import android.graphics.Shader
import android.text.style.ReplacementSpan
import android.graphics.Paint.FontMetricsInt

class GradientStrokeSpan() : CharacterStyle(),
    UpdateAppearance {
     var mStrokeWidth = 20f
    var mGradientColors: IntArray = intArrayOf(Color.BLUE, Color.RED)


    override fun updateDrawState(paint: TextPaint) {
        paint.setStyle(Paint.Style.STROKE)
        paint.strokeWidth = mStrokeWidth
        val gradient: LinearGradient = LinearGradient(
            0f, 0f, paint.measureText("Sample Text"), 0f,
            mGradientColors, null, Shader.TileMode.CLAMP
        )
        paint.shader = gradient
    }
}